# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmStubs.cpp
# Commit: 72bf9c871ded
# Full Hash: 72bf9c871ded006f56f09b24d0723365516445bb
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2023-08-05 09:19:01
# Regressor Bug: 1692065
# File Overlap Count: 5
# Description:
#   Bug 1692065 - wasm: Add tagging for 31-bit integers and i31ref. r=yury
#   
#   This commit adds a tag 0x1 for 31-bit integers in AnyRef. Any JS value
#   that is losslessly representable (without coercion) in this range is
#   stored without boxing. This is chosen to line up exactly with how the
# ==============================================================================

diff -r 28feda14fdec -r 72bf9c871ded js/src/wasm/WasmStubs.cpp
--- a/js/src/wasm/WasmStubs.cpp	Fri Aug 04 14:06:50 2023 +0000
+++ b/js/src/wasm/WasmStubs.cpp	Fri Aug 04 14:06:50 2023 +0000
@@ -1075,7 +1075,8 @@
       case ValType::Ref: {
         // Guarded against by temporarilyUnsupportedReftypeForEntry()
         MOZ_RELEASE_ASSERT(funcType.args()[i].refType().isExtern());
-        masm.branchValueConvertsToWasmAnyRefInline(scratchV, &next);
+        masm.branchValueConvertsToWasmAnyRefInline(scratchV, scratchG, scratchF,
+                                                   &next);
         masm.jump(&oolCall);
         break;
       }
@@ -1160,7 +1161,7 @@
         // have been taken care of.
         Label join;
         Label fail;
-        masm.convertValueToWasmAnyRef(src, target, &fail);
+        masm.convertValueToWasmAnyRef(src, target, scratchF, &fail);
         masm.jump(&join);
         masm.bind(&fail);
         masm.breakpoint();
@@ -2288,7 +2289,8 @@
       case ValType::Ref:
         // Guarded by temporarilyUnsupportedReftypeForExit()
         MOZ_RELEASE_ASSERT(results[0].refType().isExtern());
-        masm.convertValueToWasmAnyRef(JSReturnOperand, ReturnReg, &oolConvert);
+        masm.convertValueToWasmAnyRef(JSReturnOperand, ReturnReg,
+                                      ABINonArgDoubleReg, &oolConvert);
         GenPrintPtr(DebugChannel::Import, masm, ReturnReg);
         break;
     }